### PR TITLE
feature(engine): Using pydantic dataclasses for json-schema/validation

### PR DESCRIPTION
**Pull request by leosmerling-hopeit**
As a first step to a potential FastAPI migration,
this PR replaces the usage of `dataclasses_jsonschema` obsolete library
by the broadly adopted`pydantic`.

# BREAKING CHANGES
1. python `dataclasses.dataclass` is no longer supported in `@dataobjects` annotated classes. Pydantic `@dataclasses` (and in the future `BaseModel` should be used). Recommended way is to import from dataobjects module:
````
from hopeit.dataobjects import dataclass, dataobject, field
````
where `dataclass` and `field` are synomyms from `pydantic.dataclasses.dataclass` and `pydantic.Field`

### CODE REVIEW

**Comment by leosmerling-hopeit** [comment]
@pcanto-hopeit this needs only one more iteration for release (in 0.25.0beta1 at least):
1. Remove unusued / commented-out code

For beta2 or maybe 0.26 I will rework dataframes plugin and remove some magic from the @dataframeobject class to make it more explicit. For now is not working as expected, but since the plugin is beta that should not stop the release.

**Comment by leosmerling-hopeit** [comment]
@pcanto-hopeit : cleanup done, ready for review & test (please test simple-example and client-example apps)

File: engine/test/unit/app/test_app_config.py [hunk 1]
```diff
@@ -123,7 +123,8 @@ def _get_env_mock(var_name):
         return ENGINE_VERSION
     elif var_name == "HOPEIT_APPS_API_VERSION":
         return APPS_API_VERSION
-    raise RuntimeError(f"Missing mocked env {var_name}")
+    return None
+    # raise RuntimeError(f"Missing mocked env {var_name}")
```
Comment (pcanto-hopeit) [inline engine/test/unit/app/test_app_config.py:127]: cleanup or review?

**Review by pcanto-hopeit — APPROVED** [review]
All that remains is to unify some imports and clean up some of the lines commented in the tests. Anyway, for a beta version it's more than fine!

File: apps/examples/dataframes-example/src/dataframes_example/settings.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 """Dataframes example settings classes
 """
 
-from dataclasses import dataclass
+from hopeit.dataobjects import dataclass
 
 from hopeit.dataobjects import dataobject
```
[the remark this answers is not in the document]
Comment (pcanto-hopeit) [reply apps/examples/dataframes-example/src/dataframes_example/settings.py]: fixed in 6046139

File: engine/test/unit/app/test_app_config.py (hunk 1 above)
Comment (leosmerling-hopeit) [reply engine/test/unit/app/test_app_config.py:127]: Remove commented out line,

**Review by pcanto-hopeit — APPROVED** [review]

**Pull request merged**